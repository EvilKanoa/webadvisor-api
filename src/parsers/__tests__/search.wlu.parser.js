const path = require('path');
const util = require('util');
const readFile = util.promisify(require('fs').readFile);
const { parseXMLSearch } = require('../search.wlu.parser');

describe('WLU Course Search Parser', () => {
  it('handles empty inputs', async () => {
    expect(await parseXMLSearch('')).toBeFalsy();
    expect(await parseXMLSearch(null)).toBeFalsy();
    expect(await parseXMLSearch(false)).toBeFalsy();
  });

  it.each([
    ['search with many results', 'search.wlu.data.1.xml'],
    ['search with few results', 'search.wlu.data.2.xml'],
  ])('parses a %s correctly', async (desc, filename) => {
    const xml = await readFile(path.join(__dirname, '__data__', filename));
    const result = await parseXMLSearch(xml);
    const str = util.inspect(result, {
      depth: 10,
      maxArrayLength: Infinity,
      sorted: true,
    });
    expect(str).toMatchSnapshot();
  });
});
