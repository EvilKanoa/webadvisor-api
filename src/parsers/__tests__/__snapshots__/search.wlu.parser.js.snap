// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search.wlu.parser parses a search with few results correctly 1`] = `"[ 'CP-102', 'CP-104', 'CP-164' ]"`;

exports[`search.wlu.parser parses a search with many results correctly 1`] = `
"[ 'CP-102',
  'CP-104',
  'CP-164',
  'CP-202',
  'CP-212',
  'CP-213',
  'CP-220' ]"
`;
